src/coreclr/nativeaot/Runtime/DebugHeader.cpp

```diff
@@ -36,12 +36,12 @@ struct GlobalValueEntry
 
 // This size should be one bigger than the number of entries since a null entry
 // signifies the end of the array.
-static constexpr size_t DebugTypeEntriesArraySize = 96;
+static constexpr size_t DebugTypeEntriesArraySize = 128;
```

Comment:
Did we hit this limit? If yes, anything we can do to avoid hitting it silently?

Reply:
There are asserts but they are only in debug (it would be nice if there was a Release build assert) and are off by one (which I should have and will fixed) because they didn't account for the end/zero entry that terminates the list. The size increase was to allow for that terminating entry.  I also got carried away with the size increase; I'll reduced it.

Reply:
This should better by initialized at compile time that would solve this problem as well. We can make the structure to have the exact right size if it is initialized at compile time. It is wasteful to initialize this at runtime.

Reply:
Yes, I agree, but can we do that in a later .NET 9 PR?

Reply:
Yes, I agree it should be a separate .NET 9 PR.